src/ucm/event/event.c

```diff
@@ -658,11 +661,6 @@ ucs_status_t ucm_test_external_events(int events)
     return ucm_mmap_test_events(events & ucm_external_events, "external");
 }
 
-UCS_STATIC_INIT {
-    pthread_spin_init(&ucm_kh_lock, PTHREAD_PROCESS_PRIVATE);
```

Comment:
is it possible UCS_STATIC_CLEANUP will be called even when ucm_library_init() is never called?

Reply:
for now - no, because ```ucm_library_init``` is called from https://github.com/openucx/ucx/blob/68fa8ee661deafc826716a72be88d629e5f41f38/src/ucs/config/ucm_opts.c#L127 which is static constructor

Reply:
but from UCM side cannot rely on this, need some protection
maybe use UCS_STATIC_CLEANUP?

Reply:
> but from UCM side cannot rely on this, need some protection
> maybe use UCS_STATIC_CLEANUP?

fixed